### PR TITLE
Enable WASM extensions by default

### PR DESCRIPTION
**Pull request by jwendell**
While working on enabling WASM by default (MAISTRA-2418), an issue was found (MAISTRA-2516).
This PR addresses both issues in separate commits.

Complement of https://github.com/maistra/istio-operator/pull/747

### CODE REVIEW

**Comment by rcernich** [comment]
Does this need to include the fix for `isComponentEnabled()`?

**Comment by jwendell** [comment]
> Does this need to include the fix for `isComponentEnabled()`?

I don't think so. It's a separate issue and deserve its own Jira ticket.

**Comment by rcernich** [comment]
Maybe we change the issue to be a little broader in scope, e.g.: enable wasm extensions by default

**Comment by jwendell** [comment]
The issue is already "enable wasm extensions by default". I just think that the issue we found yesterday should be handled separated as strictly speaking it's another issue that was just discovered because of this wasm issue.

**Comment by rcernich** [comment]
it may be another issue, but without the fix, wasm extensions won't be enabled by default. :/

**Comment by jwendell** [comment]
Correct, then we could hold this one and only merge it when the other issue is fixed

File: pkg/controller/versions/strategy_v2_1.go [hunk 1]
```diff
@@ -323,6 +323,9 @@ func (v *versionStrategyV2_1) Render(ctx context.Context, cr *common.ControllerR
 		log.V(5).Info(fmt.Sprintf("rendering values:\n%s", string(rawValues)))
 	}
 
+	// Update spec.Istio back with the previous content merged with global.yaml
+	spec.Istio = v1.NewHelmValues(values)
```
Comment (rcernich) [inline pkg/controller/versions/strategy_v2_1.go:327]: We shouldn't be updating the spec.  We should be updating appliedValues or just using the values as it was before.

Comment (rcernich) [reply pkg/controller/versions/strategy_v2_1.go:327]: Ignore what I said.  It looks like there were more issues than just the isComponentEnabled() invocation (which is correct with this change), with part of the code using the merged `values` and other parts using the user specified `spec.Istio`.  Also, I verified that `spec` is not the actual resource, but the results after applying profiles.

**Review by rcernich — APPROVED** [review]

**Review by luksa — APPROVED** [review]

**Pull request merged**